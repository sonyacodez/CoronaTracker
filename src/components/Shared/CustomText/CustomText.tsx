import React from 'react';
import { useTheme } from 'styled-components';
import { IColors, IFontSizes } from '../../../types/styled';

interface IProps {
  text?: string;
  bold?: boolean;
  color?: keyof IColors;
  size?: keyof IFontSizes;
}

const CustomText: React.FC<IProps> = ({ color = 'black', size = 's14', text, bold }) => {
  const theme = useTheme();

  return (
    <span style={{ color: theme.colors[color], fontSize: theme.fontSizes[size] }}>{text}</span>
  );
};

export default CustomText;
