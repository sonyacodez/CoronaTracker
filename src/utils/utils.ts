export const storybookColorOptions = {
  black: 'black',
  white: 'white',
  red: 'red',
  green1: 'green1',
  yellow1: 'yellow1',
  purple1: 'purple1',
  turquoise1: 'turquoise1',
  lightGray1: 'lightGray1',
  gray1: 'gray1',
  lightBlue1: 'lightBlue1',
  blue: 'blue',
  navy: 'navy',
} as const;

export const storybookFontSizeOptions = {
  8: 's8',
  10: 's10',
  12: 's12',
  13: 's13',
  14: 's14',
  15: 's15',
  16: 's16',
  18: 's18',
  20: 's20',
  22: 's22',
  24: 's24',
  26: 's26',
  28: 's28',
  32: 's32',
  44: 's44',
  52: 's52',
} as const;
