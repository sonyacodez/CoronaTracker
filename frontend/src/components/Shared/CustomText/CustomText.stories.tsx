import React from 'react';
import { CustomText as CustomTextComponent } from '..';
import { withKnobs, text, select } from '@storybook/addon-knobs';
import { storybookColorOptions, storybookFontSizeOptions } from '../../../utils/utils';

export default {
  title: 'Shared',
  decorators: [withKnobs],
};

export const CustomText = () => (
  <CustomTextComponent
    text={text('Type your text here...', 'Text')}
    color={select('Colors', storybookColorOptions, storybookColorOptions.black)}
    size={select('Size', storybookFontSizeOptions, storybookFontSizeOptions[12])}
  />
);
