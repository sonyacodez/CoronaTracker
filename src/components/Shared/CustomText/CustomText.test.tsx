import React from 'react';
import CustomText from '.';
import { render } from '../../../utils/tests';
import { RenderResult } from '@testing-library/react';

describe('CustomText', () => {
  let CustomTextComponent: RenderResult;
  const sizeProps = 's12';
  const colorProps = 'black';
  const textProps = 'I am text';

  beforeEach(() => {
    CustomTextComponent = render(
      <CustomText text={textProps} />
    )
  });

  it('Render component with default styling', () => {
    const { container, getByText } = CustomTextComponent;
      expect(container.firstChild).toMatchSnapshot();
      expect(getByText(textProps)).toBeInTheDocument();
  })
  
    it('Render component with default styling', () => {
    const { container }= render(
      <CustomText text={textProps} color={colorProps} size={sizeProps} />
    )
      expect(container.firstChild).
  })
});