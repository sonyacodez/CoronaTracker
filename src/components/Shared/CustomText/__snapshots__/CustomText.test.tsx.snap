// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomText Render component with default styling 1`] = `
<span
  style="color: rgb(0, 14, 26); font-size: 1.4rem;"
>
  I am text
</span>
`;
